.circleci/config.yml

```diff
@@ -0,0 +1,235 @@
+  save_venv: &save_venv
+    save_cache:
+      key: v2-venv-{{ .Branch }}-{{ .Revision }}
+      paths:
+        - ~/venv
+  restore_venv: &restore_venv
+    restore_cache:
+      keys:
+        - v2-venv-{{ .Branch }}-{{ .Revision }}
+        - v2-venv-{{ .Branch }}
+        - v2-venv
+
+  # build steps to save/restore the marsha repository directory (to pass it between workflow steps)
+  save_repo: &save_repo
+    save_cache:
+      key: v1-repo-{{ .Branch }}-{{ .Revision }}
+      paths:
+        - .
+  restore_repo: &restore_repo
+    restore_cache:
+      key: v1-repo-{{ .Branch }}-{{ .Revision }}
+
+
+# jobs definition: they are used in ``workflows``
+jobs:
+
+  # get the code from git and save the repo to pass it to the next job
+  checkout_code:
+    <<: *python_only_config
+    steps:
+      - checkout
+      - *save_repo
+
+  # install the project code and dependencies and save the venv and pip cache
+  install_code:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_pip_cache
+      - *restore_venv
+      - run:
+          name: Install code
+          command: |
+            python -m venv ~/venv
+            source ~/venv/bin/activate
+            pip install --upgrade pip
+            pip install -e .[dev]
+      - *save_venv
+      - *save_pip_cache
+
+  # 4 next jobs are linters: mypy, black, flake8 and pylint
+  linter_mypy:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Run "mypy" linter
+          command: |
+            source ~/venv/bin/activate
+            mypy .
+
+  linter_black:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Run "black" linter
+          command: |
+            source ~/venv/bin/activate
+            black --check marsha/ 2>&1 | grep -Ev 'good job|All done'
+
+  linter_flake8:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Run "flake8" linter
+          command: |
+            source ~/venv/bin/activate
+            flake8
+
+  linter_pylint:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Run "pylint" linter
+          command: |
+            source ~/venv/bin/activate
+            pylint marsha
+
+  # django checks
+  django_checks:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Django checks
+          command: |
+            source ~/venv/bin/activate
+            django-admin.py check
+
+  # check migrations files are ok
+  check_migrations:
+    <<: *python_only_config
+    steps:
+      - *restore_repo
+      - *restore_venv
+      - run:
+          name: Check migrations
+          command: |
+            source ~/venv/bin/activate
+            django-admin.py makemigrations --check --dry-run
```

Comment:
:1st_place_medal:

Reply:
I didn't comment here but I'm not really sure it will allow us doing upgrades as we want to do them ie code base version N and N+1 both work with schema S and S+1 (this involves voluntarily delaying migrations and thus desynchronizing them from the code). But we can see this later, maybe I'm wrong.

Reply:
I propose to think about it later. I need to know more about your way to do this and also think about how to not miss migrations.

Reply:
I might be wrong, but I think this will work even with our multi-steps migration strategy.

Reply:
Yes leave it it's a good idea to check it and we'll see whaf happens with deployments.